### ISSUE TITLE
[FrameworkBundle] `setContainer()` is not being called for autowired controller services

### ISSUE THREAD

**Issue by phansys** [symfony/symfony#26554]
| Q                | A
| ---------------- | -----
| Bug report?      | yes
| Feature request? | no
| BC Break report? | no
| RFC?             | no
| Symfony version  | 3.4.6

[This](https://github.com/phansys/symfony-standard/blob/container_controller/tests/AppBundle/Controller/OtherControllerTest.php#L13) is the `symfony/standard-edition` with the reproducible error.
Am I misunderstanding #23239, or is supposed that autowired controllers should call `setContainer()`?
```
$ ./vendor/bin/simple-phpunit --filter OtherControllerTest
PHPUnit 6.5.7 by Sebastian Bergmann and contributors.

Testing Project Test Suite
E                                                                   1 / 1 (100%)

Time: 350 ms, Memory: 14.00MB

There was 1 error:

1) Tests\AppBundle\Controller\OtherControllerTest::testOther
Error: Call to a member function getParameter() on null

vendor/symfony/symfony/src/Symfony/Bundle/FrameworkBundle/Controller/Controller.php:40
src/AppBundle/Controller/DefaultController.php:18
src/AppBundle/Controller/OtherController.php:16
vendor/symfony/symfony/src/Symfony/Component/HttpKernel/HttpKernel.php:151
vendor/symfony/symfony/src/Symfony/Component/HttpKernel/HttpKernel.php:68
vendor/symfony/symfony/src/Symfony/Component/HttpKernel/Kernel.php:202
vendor/symfony/symfony/src/Symfony/Component/HttpKernel/Client.php:68
vendor/symfony/symfony/src/Symfony/Bundle/FrameworkBundle/Client.php:131
vendor/symfony/symfony/src/Symfony/Component/BrowserKit/Client.php:312
tests/AppBundle/Controller/OtherControllerTest.php:13
```

**Comment by dmaicher** [comment]
With this diff it works fine:

```diff
diff --git a/src/AppBundle/Controller/OtherController.php b/src/AppBundle/Controller/OtherController.php
index 598878e..2b527f4 100644
--- a/src/AppBundle/Controller/OtherController.php
+++ b/src/AppBundle/Controller/OtherController.php
@@ -9,10 +9,10 @@ use Symfony\Component\HttpFoundation\Request;
 class OtherController extends Controller
 {
     /**
-     * @Route("/other-path", name="homepage")
+     * @Route("/other-path", name="homepage_other")
      */
     public function indexAction(Request $request)
     {
-        return $this->get(DefaultController::class)->indexAction($request);
+        return $this->forward('AppBundle\Controller\DefaultController:indexAction');
     }
 }
```

You should never call controller actions yourself :wink: Let the Symfony kernel handle it and for example forward the request as done above.

As you can see the fix is done in the `ControllerResolver` which you are not using in your code as you call the action manually.

https://github.com/symfony/symfony/pull/23239/files#diff-9d6b7f6667ea337adcf12e1e1ba4206b

**Comment by phansys** [comment]
@dmaicher, @Tobion; I know I can create a subrequest with `Controller::forward()` (in fact, I'm skipping this "issue" with that approach by now), but I'd like to not pay the overhead to create another request at all.
I have the following use case:
My ReST API has 2 versions coexisting at same time (`Controller/V1/`, `Controller/V2`), while deprecating the old endpoints, majority of routes performs although the same action between both versions, so instead of duplicate the actions, I prefer to reuse the existing logic. To achieve this, previously I was extending new controllers from old ones. But since I'm using NelmioApiDoc in order to document my API, after migrating to version 3.0 of this bundle, I started to have notices derived from this approach (IE `Unable to merge @SWG\Get() in Controller\V2\UserController->getUserAction()`), so I decided to change the approach to something which doesn't imply to use inheritance between my controllers.
